### PR TITLE
added targetBinaryVersion to promote and tests

### PR DESCRIPTION
**Pull request by scottbommarito**
upcoming versions of the CLI support inputting targetBinaryVersion to promote, so here's the change on the VSTS extension

### CODE REVIEW

**Comment by scottbommarito** [comment]
dependent on https://github.com/Microsoft/code-push/pull/194 and https://github.com/Microsoft/code-push-service/pull/377

File: Tasks/codepush-promote/codepush-promote.js [hunk 1]
```diff
@@ -68,6 +68,7 @@ function performPromoteTask(accessKey, appName, sourceDeploymentName, targetDepl
     appName              = appName || tl.getInput("appName", true);
     sourceDeploymentName = sourceDeploymentName || tl.getInput("sourceDeploymentName", true);
     targetDeploymentName = targetDeploymentName || tl.getInput("targetDeploymentName", true);
+    appStoreVersion          = appStoreVersion || tl.getInput("appStoreVersion", false);
```
Comment (shishirx34) [inline Tasks/codepush-promote/codepush-promote.js]: format to align `=`

**Comment by geof90** [comment]
LGTM!

File: Tasks/codepush-promote/task.json [hunk 2]
```diff
@@ -127,6 +118,30 @@
             }
         },
         {
+            "name": "appStoreVersion",
+            "type": "pickList",
+            "label": "Target Binary Version",
+            "defaultValue": "Inherit",
+            "required": false,
+            "groupName": "advanced",
+            "helpMarkDown": "Semver expression that specifies the binary app version(s) this release is targetting (e.g. 1.1.0, ~1.2.3). Selecting \"Inherit\" will use the target binary version from the release being promoted.",
```
Comment (shishirx34) [inline Tasks/codepush-promote/task.json]: targeting

**Comment by shishirx34** [comment]
A typo fix, otherwise LGTM

**Comment by ryuyu** [comment]
LGTM

File: Tasks/codepush-promote/task.json [hunk 3]
```diff
@@ -103,15 +103,6 @@
             }
         },
         {
-            "name": "rollout",
```
Comment (lostintangent) [inline Tasks/codepush-promote/task.json]: I would actually prefer to leave rollout as the top input for two reasons: it's the most common field you would set as part of a promote (the other ones are pretty advanced), and it doesn't inherit. Many of the other VSTS tasks sort fields by complexity not alphabetically, so I don't think it's matters that we sort the fields by their label.

**Pull request merged**